lerobot/common/datasets/factory.py

```diff
@@ -35,11 +35,18 @@ def resolve_delta_timestamps(cfg):
                 cfg.training.delta_timestamps[key] = eval(delta_timestamps[key])
 
 
-def make_dataset(
-    cfg,
-    split="train",
-):
-    if cfg.env.name not in cfg.dataset_repo_id:
+def make_dataset(cfg, split: str = "train") -> LeRobotDataset | MultiLeRobotDataset:
+    """
+    Args:
+        cfg: A Hydra config as per the LeRobot config scheme.
+        split: TODO(now)
+    Returns:
+        The LeRobotDataset.
+    """
+    if not isinstance(cfg.dataset_repo_id, (str, ListConfig)):
+        raise ValueError("Expected cfg.dataset_repo_id to be either a single string or a list of strings.")
```

Comment:
```suggestion
        raise ValueError("Expected cfg.dataset_repo_id to be either a single string to load one dataset or a list of strings to load multiple datasets. Instead, {cfg.dataset_repo_id} provided.")
```

Reply:
If you don't mind, I won't do the last formatting bit. If it's not any of these types, something weird happened and we might get an obscure exception about the string formatting now working with the provided type.